spec-0000/index.md

```diff
@@ -29,8 +29,8 @@ All versions refer to feature releases (i.e., Python 3.8.0, NumPy 1.19.0; not Py
 
 Specifically, we recommend that:
 
-1. Support for a given version of Python be dropped **3 years** after its initial release.
-2. Support for a given version of other core packages be dropped **2 years** after their initial release.
+1. Support for Python versions be dropped **3 years** after its initial release.
+2. Support for core package dependencies be dropped **2 years** after their initial release.
```

Comment:
Thanks @max-sixty! I am trying to distill the intent of the conversation on the linked issue. I think what we're trying to say is:

(1) Limit maintenance burden, you don't need to support packages for so long.
(2) But be mindful that, if you don't support them roughly this length, other packages may struggle to install/use your library.

Maybe something like:

```suggestion
1. Support for a given Python version be dropped at latest **3 years** after its initial release.
2. Support for a given version of other core packages be dropped at latest **2 years** after their initial release.

This defines a maximum level of support, and while they may choose a much shorter window, doing so will make it harder to install their package along with others. 
```

@jarrodmillman

Reply:
Also not sure we want to even prescribe "at latest"; libraries can fully choose how long they maintain. This is almost a guidepost for where you want to aim, and beyond which you should have no guilt in dropping support.

Reply:
Thanks @stefanv 

I had this as two changes
- Clarify "support" — the original intent of this PR
- Understand whether "two years" is a suggestion / maximum / recommendation? It seems like this isn't clear and could use some more consensus building? I thought it was a good idea to have a standard, so thank you for proposing one. If the standard is just a maximum, then the standard is less useful — for example, xarray depends on pandas, so there's no point xarray taking on two years if pandas were to select one year. Maybe we need to all agree on two years, even if that's a bit long for xarray & networkx? Or maybe the standard is "two years, for inner-core packages such as pandas / scipy / matplotlib, and one year for other packages"?

Reply:
These are recommendations. So I am not sure how much detail we should provide about how you might not always follow the recommendations. I would slightly prefer to keep the recommendation that you drop it after two years.

With the understanding like for all good recommendations, that you should follow them as much as possible. But there will be times when you won't follow the recommendation. But ideally the exceptions should be minimal.

Reply:
I think that's very reasonable